ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js

```diff
@@ -205,6 +206,13 @@ User.Preferences = Backbone.AssociatedModel.extend({
     this.listenTo(this, 'change:goldenLayoutAlert', this.savePreferences)
     this.listenTo(this, 'change:mapHome', this.savePreferences)
   },
+  get(attr) {
```

Comment:
❓ so does this get function override the default get method? in other words, does this method get called every time we do `user.get(...)` ?

Reply:
this get function override the default get method for `preferences`, so when we do `user.get('user').get('preferences').get('resultFilter')`, so in this case would be the `.get('resultFilter')`

Reply:
ok, as long as it preserves the behavior of `user.get('user').get('preferences').get(...)` for the non- result filter cases, then we're good

Reply:
Yep it does, line 210 gets attributes as it usually does, and only alters data for 'resultFilter'. https://github.com/codice/ddf/blob/bc9c874dec1764ad8b41f89e80c36278fc7012ba/ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js#L210